args/physics/physics_statics.hpp

```diff
@@ -190,35 +196,228 @@ namespace args::physics
                 }
             }
 
-
+            maximumSeperation = currentMaximumSeperation;
             return false;
         }
 
-        private:
+        //---------------------------------------------------------- Polyhedron Clipping ----------------------------------------------------------------------------//
+
+        /** @brief Given a 3D plane, clips the vertices in the inputList and places the results in the output list
+         *
+         */
+        static void SutherlandHodgmanFaceClip(math::vec3& planeNormal, math::vec3& planePosition,
+            std::vector<math::vec3>& inputList, std::vector<math::vec3>& outputList)
+        {
+            for (size_t i = 0; i < inputList.size(); i++)
+            {
+                math::vec3 currentVertex = inputList.at(i);
+                math::vec3 nextVertex = i + 1 >= inputList.size() ? inputList.at(0) : inputList.at(i + 1);
+
+                bool isCurrentVertexUnderPlane = !IsPointAbovePlane(planeNormal, planePosition, currentVertex);
+                bool isNextVertexUnderPlane = !IsPointAbovePlane(planeNormal, planePosition, nextVertex);
+
+                //we always check clipping with a line that goes from the point below the plane to the point outside the plane.
+                //We do this mostly for numerical robustness reasons.
+                const math::vec3& pointAbovePlane = isCurrentVertexUnderPlane ? nextVertex : currentVertex;
+                const math::vec3& pointBelowPlane = isCurrentVertexUnderPlane ? currentVertex : nextVertex;
+
+                if (isCurrentVertexUnderPlane && isNextVertexUnderPlane)
+                {
+                    //send next vertex to outputlist
+                    outputList.push_back(nextVertex);
+                }
+                else if (!isCurrentVertexUnderPlane && isNextVertexUnderPlane)
+                {
+                    //send next vertex to outputlist and the intersection between the line and the plane
+
+                    math::vec3 intersectionPoint;
+
+                    if (FindLineToPlaneIntersectionPoint(planeNormal, planePosition,
+                        pointBelowPlane, pointAbovePlane, intersectionPoint))
+                    {
+                        outputList.push_back(intersectionPoint);
+                    }
+
+                    outputList.push_back(nextVertex);
+
+                }
+                else if (isCurrentVertexUnderPlane && !isNextVertexUnderPlane)
+                {
+                    //send intersection between the line and the plane
+
+                    math::vec3 intersectionPoint;
+
+                    if (FindLineToPlaneIntersectionPoint(planeNormal, planePosition,
+                        pointBelowPlane, pointAbovePlane, intersectionPoint))
+                    {
+                        outputList.push_back(intersectionPoint);
+                    }
+                }
+
+
+
+
+            }
```

Comment:
I expect this to be fille with _the highest quality_ code... jokes aside please watch your enter-key behaviour